storage/remote/azuread/azuread.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuread
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/google/uuid"
+)
+
+const (
+	// Clouds.
+	AzureChina      = "AzureChina"
+	AzureGovernment = "AzureGovernment"
+	AzurePublic     = "AzurePublic"
+
+	// Audiences.
+	IngestionChinaAudience      = "https://monitor.azure.cn//.default"
+	IngestionGovernmentAudience = "https://monitor.azure.us//.default"
+	IngestionPublicAudience     = "https://monitor.azure.com//.default"
```

Comment:
is the list of which of these exist online somewhere? will we have to add more in the future?

Reply:
Here is the product catalog where the backend metric store(Managed Prometheus) exists:
https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=monitor

However, these are audiences are per cloud which currently is only the three mentioned above. We will need to add if metrics datastore is made available in new clouds.